src/ui/wxWidgets/PasswordSafeFrame.cpp

```diff
@@ -2598,8 +2603,10 @@ void PasswordSafeFrame::IconizeOrHideAndLock()
   bool bModalOpen = false;
   
   if (PWSprefs::GetInstance()->GetPref(PWSprefs::UseSystemTray)) {
-    bModalOpen = (CountTopLevelWindowRecursively(this) > 1);
-    HideUI(true);
+    if (!m_sysTray->IsLocked()) {
```

Comment:
I have no objection to change condition, but outer `if` also checks for SysTray usage, so using `IsLocked` abstraction here almost changes nothing. Other code in this file also check with `m_sysTray` calls.  Looks like it would be better to check and refactor all `IsLocked`/`IsClosed` cases to use more abstract check, but, in my opinion, it should be done in other PR (this one already have a lot of changes).

Reply:
Agreed, let's keep as is.